### ISSUE TITLE
Vue 3 Support

### ISSUE THREAD

**Issue by Stovetopdropp** [shentao/vue-multiselect#1291]
Is there any plans for Vue 3 support?
Sorry if this is already being tracked somewhere I couldnt find anything on it. 

When testing out the Vue 3 beta (3.0.0-beta.15) with vue-multiselect (2.1.6) I get some compile errors

```
vue-multiselect/index.d.ts(8,36):
8:36 Type 'typeof import("/node_modules/vue/dist/vue")' is not a constructor function type.
     6 | declare class Multiselect extends Vue { }
     7 | declare class multiselectMixin extends Vue { }
  >  8 | declare class pointerMixin extends Vue { }
```

**Comment by mariusa** [comment]
Note, this is just about being usable, not porting the component to Vue 3 composition APIs.

**Comment by jehadassaf** [comment]
@shentao Now that Vue 3 is officially released, is there a plan to support it?
If there is some way we the community can help let us know.

**Comment by shentao** [comment]
@jehadassaf 
I’d love to have that, but I’m pretty overwhelmed currently with work+vuelidate rewrite for Vue 3.0. Once the latter is stable, I would want to revisit vue-multiselect in general. Currently, the project’s state is simply embarrassing as I was unable to finish the v3 rewrite that I started a long while ago.

Generally, it seems that with the composition-api, a lot of the problems in the rewrite could have been simplified. So there’s hope!

From more immediate problems that I could use some help with:
* Fixing current master so that it would be ready to release
* Evaluating how many changes would be needed to make the current master compatible with Vue 3.0

**Comment by sudip571** [comment]
@shentao 
Please make it compatible with vue3. I am eagerly waiting to use it on my vue3 project

**Comment by Brouilles** [comment]
Hi, no news? Because the most recent projects I'm working on are in Vue 3.

**Comment by DM2489** [comment]
Someone please pin this issue as it is really important for the future!

**Comment by shentao** [comment]
This is planned :)

**Comment by aperezsa-btis** [comment]
How is this issue going, it is still planned to adapt to vue3?

**Comment by mattelen** [comment]
We've just published a Vue 3 Upgrade of the multiselect component here: https://github.com/SuadeLabs/vue3-multiselect. The idea is that it is a simple replacement for this component, without needing to tweak anything. It's a good solution while this gets the official upgrade, or could be used as a base for the official upgrade.

**Comment by shentao** [comment]
Thank you @mattelen! Will look into it. Let me know if you would like to collaborate on making this official.

**Comment by mattelen** [comment]
@shentao yeah I'd be keen to collaborate! I sent you an email (to an address I found on your website) mid last week, so check that out, or let me know where to from here

**Comment by shentao** [comment]
Hey! Sorry, had the email on my list waiting, but got buried under more emails that required my attention. Just replied. Talk to you soon!

**Comment by basilbarnaby** [comment]
Any progress on this @shentao?

**Comment by shentao** [comment]
> Thanks to @mattelen, we just released v3.0.0-alpha.1 under the next tag. See https://github.com/shentao/vue-multiselect/tree/next

**Comment by mgiraldo** [comment]
i am getting this same error (`vue@3.2.44` `vite@4.1.4` `vue-multiselect@3.0.0-beta.1`)

how did folks get it to build in vue 3?
